cpp/src/arrow/CMakeLists.txt

```diff
@@ -576,6 +576,19 @@ if(ARROW_BUILD_BUNDLED_DEPENDENCIES)
   string(APPEND ARROW_PC_LIBS_PRIVATE " -larrow_bundled_dependencies")
   list(INSERT ARROW_STATIC_INSTALL_INTERFACE_LIBS 0 "Arrow::arrow_bundled_dependencies")
 endif()
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  set(GLIBCXX_USE_CXX11_ABI_SOURCE
+      ${CMAKE_CURRENT_BINARY_DIR}/try_compile_glibcxx_use_cxx_abi.cc)
+  file(WRITE ${GLIBCXX_USE_CXX11_ABI_SOURCE} "#if !_GLIBCXX_USE_CXX11_ABI\n"
+                                             "#error Not using CXX11 ABI\n" "#endif\n")
+  try_compile(IS_GLIBCXX_USE_CXX11_ABI ${CMAKE_CURRENT_BINARY_DIR}/try_compile
+              SOURCES ${GLIBCXX_USE_CXX11_ABI_SOURCE})
+  if(NOT IS_GLIBCXX_USE_CXX11_ABI)
+    string(APPEND ARROW_PC_CFLAGS " -D_GLIBCXX_USE_CXX11_ABI=0")
+  endif()
+endif()
```

Comment:
Is this really needed? We have been building and testing the R package with the dts centos 7 binary for a while now and thez worked without this? Or is this needed to use this binary on systems without dts installed?

Reply:
Yes, this is needed.
devtoolset-8 uses `_GLIBCXX_USE_CXX11_ABI=0` by default. So we don't need to specify it explicitly for our R package on CentOS 7.
But g++ on Ubuntu 18.04/20.04 uses `_GLIBCXX_USE_CXX11_ABI=1` by default. If we mix `_GLIBCXX_USE_CXX11_ABI=0` Apache Arrow C++ binaries (built on CentOS 7) and `_GLIBCXX_USE_CXX11_ABI=1` our R package, our R package can't find needed symbols. It causes a link error.

Reply:
Ah I see, thanks for the explanation!